src/ape/contracts/base.py

```diff
@@ -661,15 +661,15 @@ def deploy(self, *args, **kwargs) -> ContractInstance:
             address=receipt.contract_address,  # type: ignore
             contract_type=self.contract_type,
         )
-        self.provider.network.contract_cache[
-            contract_instance.address
-        ] = contract_instance.contract_type
+        self.chain_manager.contracts.cache_contract(
+            contract_instance.address, contract_instance.contract_type
+        )
         return contract_instance
 
 
 def _Contract(
     address: Union[str, BaseAddress, AddressType],
-    networks: "NetworkManager",
+    chain: "ChainManager",
     conversion_manager: "ConversionManager",
```

Comment:
Having to pass these through is annoying, would be nice to add this as a method to `ContractCache` instead to leverage `BaseManager`